stabbity.lic

```diff
@@ -0,0 +1,253 @@
+        waitrt?
+        bput(recover_command, 'You pick up', 'suddenly leaps')
+        return if npc_dead?
+        waitrt?
+      when 'There is nothing else'
+        @npc_is_alive = false
+        return
+      end
+    end
+  end
+
+  def get_npcs
+    DRRoom.npcs - @ignore_npcs
+  end
+
+  def npc_dead?
+    if DRRoom.dead_npcs.include?(@target) || get_npcs.empty?
+      echo "*** Target #{@target} has died ***" if @debug
+      @npc_is_alive = false
+      return true
+    else
+      return false
+    end
+  end
+
+  def using_right_weapon?
+    if @alternate_weapon_npcs.include? @target
+      echo "*** Switching to alternate weapon ***" if @debug
+      use_weapon 'alternate'
+      hide?
+    elsif @thrown_weapon_npcs.include? @target
+      echo "*** Switching to thrown weapon ***" if @debug
+      use_weapon 'thrown'
+    elsif @current_weapon != 'preferred'
+      echo "*** Switching to preferred weapon" if @debug
+      use_weapon
+      hide?
+    end
+  end
+
+  def use_weapon(type = 'preferred')
+    echo "*** Already on right type ***" if @debug && @current_weapon == type
+    return if @current_weapon == type
+    case type
+    when 'preferred'
+      @current_weapon = 'preferred'
+      return if right_hand =~ /#{@preferred_weapon}/
+      @equipmanager.stow_weapon
+      @equipmanager.wield_weapon(@preferred_weapon) unless @preferred_weapon.empty?
+    when 'alternate'
+      @current_weapon = 'alternate'
+      return if right_hand =~ /#{@alternate_weapon}/
+      @equipmanager.stow_weapon
+      @equipmanager.wield_weapon(@alternate_weapon) unless @alternate_weapon.empty?
+    when 'thrown'
+      @current_weapon = 'thrown'
+      return if right_hand =~ /#{@thrown_weapon}/
+      @equipmanager.stow_weapon
+      @equipmanager.wield_weapon(@thrown_weapon)
+    end
+    echo "*** Changed weapon type to #{type} ***" if @debug
+  end
+
+  def watch_target
+    case bput("watch #{@target}", 'you could try to pedal', 
+                                 'you could try to bob', 
+                                 'you could try to duck', 
+                                 'you could try to jump', 
+                                 'you could try to lean', 
+                                 'you could try to cower',
+                                 '.*')
+    when 'you could try to pedal'
+      @arena_trap = 'pedal'
+    when 'you could try to bob'
+      @arena_trap = 'bob'
+    when 'you could try to duck'
+      @arena_trap = 'duck'
+    when 'you could try to jump'
+      @arena_trap = 'jump'
+    when 'you could try to lean'
+      @arena_trap = 'lean'
+    when 'you could try to cower'
+      @arena_trap = 'cower'
+    end
+  end
+
+  def dodge_arena_trap
+    if @arena_trap
+      echo "*** Performing arena action #{@arena_trap} ***" if @debug
+      bput(@arena_trap, '.*')
+      @arena_trap = nil
+    end
+  end
+
+  def loot_mob
+    bput('loot', '.*')
+    pause 1
+    #todo/maybe -- pick up loot
```

Comment:
You might want to look at combat-trainer and see if you can steal any looting logic there.

As a user of this script, I would except it to loot the same way.

Reply:
I thought about working to untangle LootProcess from combat-trainer, but that also requires untangling the whole base GameState too, which just seems... unattractive. So, I could implement a more limited form of looting (coins, boxes, gems, and skins) which in the context of the intended purpose of this script seems ok. Thoughts?

Reply:
@rpherbig ^

Reply:
New baby incoming, can't process that. @Sarvatt @jandersson (or anyone else) can you provide some guidance?

Reply:
Theres been discussion about breaking out the classes into individual files in which case you could possibly import the lootprocess class and gamestate class and make use of it?

Reply:
@jandersson I was considering that approach, but then I also need to implement the same model `execute` loop as combat trainer for the process to work, which requires a lot more rework, and it seems like it will introduce far more complexity than desired to just add loot support.